### PR TITLE
Adding handling for Serverless Apps into the resource explorer

### LINKED ISSUE #398: [0.5/2] Display all existing CloudFormation stacks in Explorer
**Is your feature request related to a problem? Please describe.**

User should be able to see a listing of all current CloudFormation stacks in the AWS Explorer

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kiiadi**
Fixes #398

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Testing

New unit tests added

## Screenshots (if appropriate)
screen shot 2018-11-07 at 1 36 12 am

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    private val client: CloudFormationClient = AwsClientManager.getInstance(project).getClient()
+
+    override fun loadResources(paginationToken: String?): Collection<AwsExplorerNode<*>> {
+        val response = client.describeStacks { request ->
+            paginationToken?.let { request.nextToken(it) }
+        }
+
+        val nodes = response.stacks().filterNotNull().filter { stack ->
+            client.describeStackResources { it.stackName(stack.stackName()) }.stackResources().any { it.resourceType() == "AWS::Lambda::Function" }
```
Comment (awschristou) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: Consider putting `AWS::Lambda::Function` into a constant somewhere that can be used throughout the extension

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsExplorerNode.kt [hunk 2]
```diff
@@ -64,19 +61,28 @@ abstract class AwsExplorerServiceRootNode(project: Project, value: String) :
...
     }
 
-    override fun getChildren(): MutableList<AwsExplorerNode<*>> = childNodes
-
-    fun loadData(paginationToken: String? = null): Collection<AwsExplorerNode<*>> = try {
+    internal fun loadData(paginationToken: String? = null): Collection<AwsExplorerNode<*>> = try {
         loadResources(paginationToken)
     } catch (e: Exception) {
-        LOG.warn("Failed to load explorer nodes", e)
+        handleException("Failed to load explorer nodes", e)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsExplorerNode.kt]: Not seeing the benefit of this change

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: Currently its defined in `software.aws.toolkits.jetbrains.services.cloudformation.ResourcesKt#LAMBDA_FUNCTION_TYPE`

Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt]: 👍 this was definitely just me being lazy

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessApplicationsNodeTest.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import software.aws.toolkits.jetbrains.utils.delegateMock
+
+class ServerlessApplicationsNodeTest {
+
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    private val mockClient by lazy {
+        val mockClientManager = AwsClientManager.getInstance(projectRule.project) as MockClientManager
```
[the remark this answers is not in the document]
Comment (kiiadi) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessApplicationsNodeTest.kt]: 👍

**Pull request merged**